src/components/calcite-combobox/utils.ts

```diff
@@ -24,12 +25,17 @@ export function hasActiveChildren(node: HTMLCalciteComboboxItemElement): boolean
 }
 
 export function getDepth(element: HTMLElement): number {
-  const [parent, grandparent] = getAncestors(element);
-  if (!parent) {
+  if (!Build.isBrowser) {
     return 0;
   }
-  if (!grandparent) {
-    return 1;
-  }
-  return 2;
+
+  const result = document.evaluate(
```

Comment:
wow this is an advanced maneuver 🕶️ ⚡

Reply:
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia.giphy.com%2Fmedia%2FgkrC5xECBbAbe%2Fgiphy.gif&f=1&nofb=1)

Reply:
Is xpath use ok @jcfranco ?

Reply:
TBH, I need a refresher on it. The last time I dealt w/ xpath was in the good ol' Flex/ActionScript days. 😂  

How does it deal in terms of performance and does it work across shadow DOM boundaries?

Reply:
I'm not sure on the performance but it seems to work fine.

It shouldn't need to work across shadow for this use case but I think it does.